### DISCUSSION TITLE
Added GitInfo, build fails The specified task executable "sh" could not be run

### DISCUSSION THREAD

**Discussion by sandrock** [devlooped/GitInfo#327] [Q&A] [answered]
Hi,

I'm trying to add `GitInfo` to my [mynatime project](https://github.com/sandrock/mynatime). My setup is Manjaro Linux + Rider.

Build fails with this obscure error:

> 0>------- Started building project: Mynatime.Infrastructure
/usr/bin/sh: line 1: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
/usr/bin/sh: line 1: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
0>GitInfo.targets(339,5): Error MSB6003 : The specified task executable "sh" could not be run. System.IO.DirectoryNotFoundException: The working directory "/usr/bin/sh: line 1: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8);/home/sandrock/dev/github/sandrock/mynatime/" does not exist.
   at Microsoft.Build.Tasks.Exec.GetWorkingDirectory()
   at Microsoft.Build.Utilities.ToolTask.GetProcessStartInfo(String pathToTool, String commandLineCommands, String responseFileSwitch)
   at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
   at Microsoft.Build.Tasks.Exec.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
   at Microsoft.Build.Utilities.ToolTask.Execute()
0>------- Finished building project: Mynatime.Infrastructure. Succeeded: False. Errors: 1. Warnings: 0
Build completed in 00:00:00.678

Nugets were added by Rider. Here is the new entry in csproj:

```
  <ItemGroup>
    <PackageReference Include="GitInfo" Version="3.3.3">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="6.0.2" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
  </ItemGroup>
```

I'm not sure where to look for resolution. 

```
dotnet --list-sdks
2.2.207 [/usr/share/dotnet/sdk]
3.1.120 [/usr/share/dotnet/sdk]
6.0.124 [/usr/share/dotnet/sdk]
7.0.113 [/usr/share/dotnet/sdk]
```

**Comment by sandrock** [comment]
Maybe related to #174

**Comment by spmason** [comment]
Logging in to say that I think the problem is this issue: https://github.com/dotnet/msbuild/issues/4194

Because of this log line:
```
The working directory "/usr/bin/sh: line 1: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8);/home/sandrock/dev/github/sandrock/mynatime/" does not exist.
```

ie somewhere GitInfo is deciding that the current working directory is `/usr/bin/sh: line 1: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8);/home/sandrock/dev/github/sandrock/mynatime/` - due to that msbuild issue

The fix should be to install the correct Locale on your build machine(s) - or use a newer msbuild perhaps..

**Comment by sandrock** [comment, marked as answer]
My `/etc/locale.gen` was:

```
en_GB.UTF-8 UTF-8
```

I changed it to:

```
en_GB.UTF-8 UTF-8
en_US.UTF-8 UTF-8
C.UTF-8 UTF-8
```

Did `sudo locale-gen` and `sudo localectl set-locale LANG=en_GB.utf8`

Problem solved.
